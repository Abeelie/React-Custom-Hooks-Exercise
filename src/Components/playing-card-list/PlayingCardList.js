import React from "react";
import { v4 as uuid } from "uuid";
import PlayingCard from "../playing-card/PlayingCard";
import "./PlayingCardList.css";
import { useAxios } from "../../hooks/useAxios";

/* Renders a list of playing cards.
 * Can also add a new card at random. */
function CardTable() {
  const [data, setData, clearData] = useAxios("https://deckofcardsapi.com/api/deck/new/draw/");

  return (
    <div className="PlayingCardList">
      <h3>Pick a card, any card!</h3>
      <div>
        <button onClick={setData}>Add a playing card!</button>
        <button onClick={clearData}>Clear All Cards</button>
      </div>
      <div className="PlayingCardList-card-area">
        {data.map(cardData => (
          <PlayingCard key={uuid()} front={cardData.cards[0].image} />
        ))}
      </div>
    </div>
  );
}

CardTable.defaultProps = {};

export default CardTable;
