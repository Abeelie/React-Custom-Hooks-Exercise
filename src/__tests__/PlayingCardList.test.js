import React from "react";
import { render } from "@testing-library/react";
import PlayingCardList from "../Components/playing-card-list/PlayingCardList";


it("renders without crashing", function() {
  render(<PlayingCardList />);
});

it("matches snapshot", function() {
  const { asFragment } = render(<PlayingCardList />);
  expect(asFragment()).toMatchSnapshot();
});