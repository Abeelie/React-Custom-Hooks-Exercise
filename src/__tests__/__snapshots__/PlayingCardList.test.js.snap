// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="PlayingCardList"
  >
    <h3>
      Pick a card, any card!
    </h3>
    <div>
      <button>
        Add a playing card!
      </button>
      <button>
        Clear All Cards
      </button>
    </div>
    <div
      class="PlayingCardList-card-area"
    />
  </div>
</DocumentFragment>
`;
